bootstrap-salt.sh

```diff
@@ -1813,27 +1819,26 @@ __function_defined() {
 #                 a boot process, such as on AWS AMIs. This func will wait until the boot
 #                 process is finished so the script doesn't exit on a locked proc.
 #----------------------------------------------------------------------------------------------------------------------
+APT_ERR=$(mktemp /tmp/apt_error.XXXX)
 __wait_for_apt(){
     # Timeout set at 15 minutes
     WAIT_TIMEOUT=900
 
     # Run our passed in apt command
-    "${@}"
+    "${@}" 2>$APT_ERR
     APT_RETURN=$?
 
-    # If our exit code from apt is 100, then we're waiting on a lock
-    while [ $APT_RETURN -eq 100 ]; do
+    # Make sure we're not waiting on a lock
+    while [ $APT_RETURN -ne 0 ] && grep -q -v '^E: Could not get lock' "$APT_ERR"; do
```

Comment:
It appears that `grep` with `-v` always returns `0`. And this is really not that we expect. We need to keep trying only on non-zero code from `apt` and the lock message appeared. Please omit `-v`, I misunderstood the intended behavior.

Reply:
I think this is still correct? Without `-v` we'd return an exit code of `0` from `grep` if we found the string 'E: Could not get lock'. We want the inverse of that.

Reply:
Fixed.